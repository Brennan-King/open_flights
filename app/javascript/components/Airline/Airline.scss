.airline-container {
  margin-left: auto;
  margin-right: auto;
  display: grid;
  grid-template-columns: repeat(2, 1fr);
}

.airline-column {
  background: #fff;
  height: 100vh;
  overflow: auto;
}

.airline-header-container {
  padding-left: 50px;
}

::-webkit-scrollbar {
  display: none;
}