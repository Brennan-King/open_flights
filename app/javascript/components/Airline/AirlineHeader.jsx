import React from 'react';
import PropTypes from 'prop-types';
import './AirlineHeader.scss';

const propTypes = {
  /** Object containing an airline's attributes. */
  airlineAttributes: PropTypes.object,
};

const propDefaults = {
  airlineAttributes: {},
};

/**
 * Constructs an airline header component used to render a header
 * for a specific airline.
 * @param {object} props the props used to configure the airline header
 * @return {node} a div containing the configured airline header
 */
const AirlineHeader = (props) => {
  const {airlineAttributes} = props;
  const airlineName = airlineAttributes.name;
  const airlineImageUrl = airlineAttributes.image_url;
  const airlineAverageScore = airlineAttributes.avg_score;
  const totalAirlineReviews = airlineAttributes.reviews ?
    airlineAttributes.reviews.length : 0;
  const totalAirlineReviewsText = `${totalAirlineReviews} user reviews`;
  const airlineRatingOutOfFiveText = `${airlineAverageScore} out of 5`;

  return (
    <div className="airline-header-container">
      <h1 className="airline-header">
        <img
          className="airline-header-image"
          src={airlineImageUrl}
          alt={airlineName} />
        {airlineName}
      </h1>
      <div>
        <div className="airline-total-reviews">
          {airlineRatingOutOfFiveText}
        </div>
        <div className="airline-rating-out-of-five">
          {totalAirlineReviewsText}
        </div>
      </div>
    </div>
  );
};

AirlineHeader.propDefaults = propDefaults;
AirlineHeader.propTypes = propTypes;
export default AirlineHeader;
