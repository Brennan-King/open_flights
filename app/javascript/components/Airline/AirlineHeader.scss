.airline-header-container {
  padding: 50px 100px 50px 0;
  font-size: 30px;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
}

.airline-header-image {
  height: 60px;
  width: 60px;
  border-radius: 100%;
  border: 1px solid darkgray;
  margin-bottom: -8px;
  margin-right: 5px;
}

.airline-total-reviews {
  font-family: Verdana, Geneva, Tahoma, sans-serif;
}

.airline-rating-out-of-five {
  font-size: 18px;
  font-weight: bold;
  padding: 10px 0;
  font-family: Verdana, Geneva, Tahoma, sans-serif;
}