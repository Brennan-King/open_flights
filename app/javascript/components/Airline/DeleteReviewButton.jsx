import React from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import './DeleteReviewButton.scss';

const propTypes = {
  /** The id of the review */
  reviewId: PropTypes.number,
  /** Callback function to set delete button state */
  setDeleteButtonClicked: PropTypes.func,
};

const propDefaults = {
  reviewId: 0,
  setDeleteButtonClicked: () => {},
};

/**
 * Constructs a delete button component used to delete a specific review.
 * @param {object} props the props used to configure the component
 * @return {node} a div containing the configured delete button
 */
const DeleteReviewButton = (props) => {
  const {reviewId, setDeleteButtonClicked} = props;

  const onDeleteReviewButtonClick = () => {
    const csrfToken = document.querySelector('[name=csrf-token]').content;
    axios.defaults.headers.common['X-CSRF-TOKEN'] = csrfToken;

    axios.delete(`/api/v1/reviews/${reviewId}`);
    setDeleteButtonClicked(true);
  };

  return (
    <div
      className="delete-review-button"
      onClick={() => onDeleteReviewButtonClick()}
    >
      <div className="x-symbol">x</div>
    </div>
  );
};

DeleteReviewButton.propTypes = propTypes;
DeleteReviewButton.propDefaults = propDefaults;
export default DeleteReviewButton;
