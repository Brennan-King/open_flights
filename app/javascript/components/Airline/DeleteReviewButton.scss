.delete-review-button {
  height: 23px;
  width: 18px;
  background-color: red;
  margin-left: auto;
  cursor: pointer;
  transition: ease-in-out 0.1s;
  border: 2px solid black;
  box-shadow: 2px 2px 2px grey;
}

.x-symbol {
  margin-top: -10px;
  color: white;
  font-weight: lighter;
}

.delete-review-button:hover {
  background: darkred;
}