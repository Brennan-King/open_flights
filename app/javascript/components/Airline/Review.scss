.review-card {
  border: 2px solid black;
  border-radius: 4px;
  padding: 20px;
  margin: 0 0 20px 0;
  box-shadow: 7px 7px 7px lightgray;
}

.review-rating-container {
  display: flex;
  flex-direction: row;
}

.review-title {
  padding: 20px 0 0 0;
  font-size: 18px;
}

.review-description {
  padding: 0 0 20px 0;
  font-size: 14px;
}