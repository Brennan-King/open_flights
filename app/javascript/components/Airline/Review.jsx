import React from 'react';
import PropTypes from 'prop-types';
import AirlineRating from '../AirlineRating/AirlineRating';
import DeleteReviewButton from './DeleteReviewButton';
import reviewIcon from '../../images/reviewIcon.png';
import './Review.scss';

const propTypes = {
  /** The title of the review */
  reviewTitle: PropTypes.string,
  /** The description of the review */
  reviewDescription: PropTypes.string,
  /** The score of the review */
  reviewScore: PropTypes.number,
  /** The id of the review */
  reviewId: PropTypes.number,
  /** Callback function to set delete button state */
  setDeleteButtonClicked: PropTypes.func,
};

const propDefaults = {
  reviewTitle: '',
  reviewDescription: '',
  reviewScore: 0,
  reviewId: 0,
  setDeleteButtonClicked: () => {},
};

/**
 * Constructs a review component used to display a specific review.
 * @param {object} props the props used to configure the review
 * @return {node} a div containing the configured review
 */
const Review = (props) => {
  const {reviewTitle,
    reviewDescription,
    reviewScore,
    reviewId,
    setDeleteButtonClicked,
  } = props;

  return (
    <div className="review-card">
      <div className="review-rating-container">
        <img
          className="review-icon"
          src={reviewIcon}
          alt="reviewIcon"
        />
        <AirlineRating
          reviewScore={reviewScore}
        />
        <DeleteReviewButton
          reviewId={reviewId}
          setDeleteButtonClicked={setDeleteButtonClicked}
        />
      </div>
      <div className="review-title">{reviewTitle}</div>
      <div className="review-description">{reviewDescription}</div>
    </div>
  );
};

Review.propTypes = propTypes;
Review.propDefaults = propDefaults;
export default Review;
