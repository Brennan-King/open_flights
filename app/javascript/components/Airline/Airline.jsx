import React, {useState, useEffect} from 'react';
import axios from 'axios';
import AirlineHeader from './AirlineHeader';
import ReviewForm from './ReviewForm';
import Review from './Review';
import './Airline.scss';

/**
 * Constructs an airline component used to display a specific airline.
 * @param {object} props the props used to configure the airline
 * @return {node} a div containing the configured airline
 */
const Airline = (props) => {
  const [airline, setAirline] = useState({});
  const [reviewFormInput, setReviewFormInput] = useState({});
  const [reviewFormRatingSelection, setReviewFormRatingSelection] = useState();

  useEffect(() => {
    // eslint-disable-next-line react/prop-types
    const urlSlug = props.match.params.slug;
    const airlinesEndpoint = `/api/v1/airlines/${urlSlug}`;

    axios.get(airlinesEndpoint).then( (response) => setAirline(response.data))
        .catch( (response) => console.log(response));
  }, [reviewFormRatingSelection]);

  const onReviewFormInputChange = (event) => {
    event.preventDefault();
    setReviewFormInput(Object.assign({}, reviewFormInput,
        {[event.target.name]: event.target.value}));
  };

  const onReviewFormSubmit = (event) => {
    event.preventDefault();

    const csrfToken = document.querySelector('[name=csrf-token]').content;
    axios.defaults.headers.common['X-CSRF-TOKEN'] = csrfToken;

    const review = {
      title: reviewFormInput.title,
      description: reviewFormInput.description,
      score: reviewFormRatingSelection,
      airline_id: airline.id,
    };

    axios.post('/api/v1/reviews', {review});
    setReviewFormRatingSelection(0);
  };

  const getAirlineReviews = () => {
    let airlineReviews;
    if (airline.reviews) {
      airlineReviews = airline.reviews.map((airlineReview, index) => {
        return (
          <Review
            key={index}
            reviewTitle={airlineReview.title}
            reviewDescription={airlineReview.description}
            reviewScore={airlineReview.score}
          />
        );
      });
    }

    return airlineReviews;
  };

  return (
    <div className="airline-container">
      <div className="airline-column">
        <div className="airline-header-container">
          {
            !!airline && <AirlineHeader airlineAttributes={airline}/>
          }
          {getAirlineReviews()}
        </div>
      </div>
      <div className="airline-column">
        <div>
          <ReviewForm
            onReviewFormInputChange={onReviewFormInputChange}
            onReviewFormSubmit={onReviewFormSubmit}
            setReviewFormRating={setReviewFormRatingSelection}
            airlineAttributes={airline}
          />
        </div>
      </div>
    </div>
  );
};

export default Airline;
