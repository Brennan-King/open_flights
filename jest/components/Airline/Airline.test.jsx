/* eslint-disable max-len */
import 'jsdom-global/register';
import React from 'react';
import Airline from '../../../app/javascript/components/Airline/Airline';
import {configure, mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import {act} from 'react-dom/test-utils';
import * as axios from 'axios';

configure({adapter: new Adapter()});
jest.mock('axios');

/**
 * Test suite for Airline component
 */
describe('Airline', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render an Airline component when the service successfully returns data', async () => {
    let component;
    axios.get.mockImplementation(() => Promise.resolve({data: {
      id: '123',
      name: 'testAirlineName',
      image_url: 'testAirlineImageUrl',
      avg_score: 4.5,
      reviews: [{
        title: 'testReviewTitle',
        description: 'testReviewDescription',
        score: 3,
        id: '456',
      }],
    }}));

    await act(async () => {
      component = mount(
          <Airline
            match={{params: {slug: {}}}}
          />,
      );
    });
    await component.update();

    expect(component).toMatchSnapshot();
  });

  it('should render an Airline component when the service unsuccessfully returns data', async () => {
    let component;
    axios.get.mockImplementation(() => Promise.reject(new Error('Service call failed')));

    await act(async () => {
      component = mount(
          <Airline
            match={{params: {slug: {}}}}
          />,
      );
    });
    await component.update();

    expect(component).toMatchSnapshot();
  });

  it('should successfully submit a review', async () => {
    let component;
    axios.get.mockImplementation(() => Promise.resolve({data: {
      id: '123',
      name: 'testAirlineName',
      image_url: 'testAirlineImageUrl',
      avg_score: 4.5,
      reviews: [{
        title: 'testReviewTitle',
        description: 'testReviewDescription',
        score: 3,
        id: '456',
      }],
    }}));

    await act(async () => {
      component = mount(
          <Airline
            match={{params: {slug: {}}}}
          />,
      );
    });
    await component.update();

    component.find('#review-form-title-input').simulate('change', {target: {value: 'Title', name: 'title'}});
    component.find('#review-form-description-input').simulate('change', {target: {value: 'Description', name: 'description'}});
    component.find('#review-form').simulate('submit');
    expect(component).toMatchSnapshot();
  });
});
