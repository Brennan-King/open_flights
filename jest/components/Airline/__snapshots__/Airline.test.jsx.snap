// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Airline should render an Airline component when the service successfully returns data 1`] = `
<Airline
  match={
    Object {
      "params": Object {
        "slug": Object {},
      },
    }
  }
>
  <div
    className="airline-container"
  >
    <div
      className="airline-column"
    >
      <div
        className="airline-header-container"
      >
        <AirlineHeader
          airlineAttributes={
            Object {
              "avg_score": 4.5,
              "id": "123",
              "image_url": "testAirlineImageUrl",
              "name": "testAirlineName",
              "reviews": Array [
                Object {
                  "description": "testReviewDescription",
                  "id": "456",
                  "score": 3,
                  "title": "testReviewTitle",
                },
              ],
            }
          }
        >
          <div
            className="airline-header-container"
          >
            <h1>
              <img
                alt="testAirlineName"
                className="airline-header-image"
                src="testAirlineImageUrl"
              />
              testAirlineName
            </h1>
            <div>
              <div
                className="airline-total-reviews"
              >
                4.5 out of 5
              </div>
              <div
                className="airline-rating-out-of-five"
              >
                1 user reviews
              </div>
            </div>
          </div>
        </AirlineHeader>
        <Review
          key="0"
          reviewDescription="testReviewDescription"
          reviewId="456"
          reviewScore={3}
          reviewTitle="testReviewTitle"
          setDeleteButtonClicked={[Function]}
        >
          <div
            className="review-card"
          >
            <div
              className="review-rating-container"
            >
              <img
                alt="reviewIcon"
                className="review-icon"
                src="test-file-stub"
              />
              <AirlineRating
                reviewScore={3}
              >
                <span
                  className="airline-rating-wrapper"
                >
                  <div
                    className="airline-rating-filled-in"
                  />
                  <div
                    className="airline-rating-filled-in"
                  />
                  <div
                    className="airline-rating-filled-in"
                  />
                  <div
                    className="airline-rating-not-filled-in"
                  />
                  <div
                    className="airline-rating-not-filled-in"
                  />
                </span>
              </AirlineRating>
              <DeleteReviewButton
                reviewId="456"
                setDeleteButtonClicked={[Function]}
              >
                <div
                  className="delete-review-button"
                  id="delete-review-button"
                  onClick={[Function]}
                >
                  <div
                    className="x-symbol"
                  >
                    x
                  </div>
                </div>
              </DeleteReviewButton>
            </div>
            <div
              className="review-title"
            >
              testReviewTitle
            </div>
            <div
              className="review-description"
            >
              testReviewDescription
            </div>
          </div>
        </Review>
      </div>
    </div>
    <div
      className="airline-column"
    >
      <div>
        <ReviewForm
          airlineAttributes={
            Object {
              "avg_score": 4.5,
              "id": "123",
              "image_url": "testAirlineImageUrl",
              "name": "testAirlineName",
              "reviews": Array [
                Object {
                  "description": "testReviewDescription",
                  "id": "456",
                  "score": 3,
                  "title": "testReviewTitle",
                },
              ],
            }
          }
          onReviewFormInputChange={[Function]}
          onReviewFormSubmit={[Function]}
          setReviewFormRating={[Function]}
        >
          <div
            className="review-form-container"
          >
            <form
              id="review-form"
              onSubmit={[Function]}
            >
              <div
                className="review-form-headline"
              >
                Have and experience  with testAirlineName? Share your review!
              </div>
              <div
                className="review-form-inputs-container"
              >
                <input
                  className="review-form-title-input"
                  id="review-form-title-input"
                  name="title"
                  onChange={[Function]}
                  placeholder="Review Title"
                  type="text"
                />
              </div>
              <div
                className="review-form-inputs-container"
              >
                <input
                  className="review-form-description-text-area"
                  id="review-form-description-input"
                  name="description"
                  onChange={[Function]}
                  placeholder="Review Description"
                  type="text"
                />
              </div>
              <div
                className=""
              >
                <div
                  className="review-form-rating-container"
                >
                  <div
                    className="review-form-rating-title-text"
                  >
                    Rate This Airline
                  </div>
                  <div
                    className="review-form-rating-options"
                  >
                    <input
                      checked={false}
                      className="review-form-rating-options-input"
                      id="rating-5"
                      name="rating"
                      type="radio"
                      value={5}
                    />
                    <label
                      className="review-form-rating-options-label"
                      id="review-form-rating-label"
                      onClick={[Function]}
                    />
                    <input
                      checked={false}
                      className="review-form-rating-options-input"
                      id="rating-4"
                      name="rating"
                      type="radio"
                      value={4}
                    />
                    <label
                      className="review-form-rating-options-label"
                      id="review-form-rating-label"
                      onClick={[Function]}
                    />
                    <input
                      checked={false}
                      className="review-form-rating-options-input"
                      id="rating-3"
                      name="rating"
                      type="radio"
                      value={3}
                    />
                    <label
                      className="review-form-rating-options-label"
                      id="review-form-rating-label"
                      onClick={[Function]}
                    />
                    <input
                      checked={false}
                      className="review-form-rating-options-input"
                      id="rating-2"
                      name="rating"
                      type="radio"
                      value={2}
                    />
                    <label
                      className="review-form-rating-options-label"
                      id="review-form-rating-label"
                      onClick={[Function]}
                    />
                    <input
                      checked={false}
                      className="review-form-rating-options-input"
                      id="rating-1"
                      name="rating"
                      type="radio"
                      value={1}
                    />
                    <label
                      className="review-form-rating-options-label"
                      id="review-form-rating-label"
                      onClick={[Function]}
                    />
                  </div>
                </div>
              </div>
              <button
                className="review-form-submit-button"
                type="submit"
              >
                Submit Your Review
              </button>
              <div
                className="review-form-airplane-image-container"
              >
                <img
                  alt="reviewFormAirplaneImage"
                  className="review-form-airplane-image"
                  src="test-file-stub"
                />
              </div>
            </form>
          </div>
        </ReviewForm>
      </div>
    </div>
  </div>
</Airline>
`;

exports[`Airline should render an Airline component when the service unsuccessfully returns data 1`] = `
<Airline
  match={
    Object {
      "params": Object {
        "slug": Object {},
      },
    }
  }
>
  <div
    className="airline-container"
  >
    <div
      className="airline-column"
    >
      <div
        className="airline-header-container"
      >
        <AirlineHeader
          airlineAttributes={Object {}}
        >
          <div
            className="airline-header-container"
          >
            <h1>
              <img
                className="airline-header-image"
              />
            </h1>
            <div>
              <div
                className="airline-total-reviews"
              >
                undefined out of 5
              </div>
              <div
                className="airline-rating-out-of-five"
              >
                0 user reviews
              </div>
            </div>
          </div>
        </AirlineHeader>
      </div>
    </div>
    <div
      className="airline-column"
    >
      <div>
        <ReviewForm
          airlineAttributes={Object {}}
          onReviewFormInputChange={[Function]}
          onReviewFormSubmit={[Function]}
          setReviewFormRating={[Function]}
        >
          <div
            className="review-form-container"
          >
            <form
              id="review-form"
              onSubmit={[Function]}
            >
              <div
                className="review-form-headline"
              >
                Have and experience  with undefined? Share your review!
              </div>
              <div
                className="review-form-inputs-container"
              >
                <input
                  className="review-form-title-input"
                  id="review-form-title-input"
                  name="title"
                  onChange={[Function]}
                  placeholder="Review Title"
                  type="text"
                />
              </div>
              <div
                className="review-form-inputs-container"
              >
                <input
                  className="review-form-description-text-area"
                  id="review-form-description-input"
                  name="description"
                  onChange={[Function]}
                  placeholder="Review Description"
                  type="text"
                />
              </div>
              <div
                className=""
              >
                <div
                  className="review-form-rating-container"
                >
                  <div
                    className="review-form-rating-title-text"
                  >
                    Rate This Airline
                  </div>
                  <div
                    className="review-form-rating-options"
                  >
                    <input
                      checked={false}
                      className="review-form-rating-options-input"
                      id="rating-5"
                      name="rating"
                      type="radio"
                      value={5}
                    />
                    <label
                      className="review-form-rating-options-label"
                      id="review-form-rating-label"
                      onClick={[Function]}
                    />
                    <input
                      checked={false}
                      className="review-form-rating-options-input"
                      id="rating-4"
                      name="rating"
                      type="radio"
                      value={4}
                    />
                    <label
                      className="review-form-rating-options-label"
                      id="review-form-rating-label"
                      onClick={[Function]}
                    />
                    <input
                      checked={false}
                      className="review-form-rating-options-input"
                      id="rating-3"
                      name="rating"
                      type="radio"
                      value={3}
                    />
                    <label
                      className="review-form-rating-options-label"
                      id="review-form-rating-label"
                      onClick={[Function]}
                    />
                    <input
                      checked={false}
                      className="review-form-rating-options-input"
                      id="rating-2"
                      name="rating"
                      type="radio"
                      value={2}
                    />
                    <label
                      className="review-form-rating-options-label"
                      id="review-form-rating-label"
                      onClick={[Function]}
                    />
                    <input
                      checked={false}
                      className="review-form-rating-options-input"
                      id="rating-1"
                      name="rating"
                      type="radio"
                      value={1}
                    />
                    <label
                      className="review-form-rating-options-label"
                      id="review-form-rating-label"
                      onClick={[Function]}
                    />
                  </div>
                </div>
              </div>
              <button
                className="review-form-submit-button"
                type="submit"
              >
                Submit Your Review
              </button>
              <div
                className="review-form-airplane-image-container"
              >
                <img
                  alt="reviewFormAirplaneImage"
                  className="review-form-airplane-image"
                  src="test-file-stub"
                />
              </div>
            </form>
          </div>
        </ReviewForm>
      </div>
    </div>
  </div>
</Airline>
`;

exports[`Airline should successfully submit a review 1`] = `
<Airline
  match={
    Object {
      "params": Object {
        "slug": Object {},
      },
    }
  }
>
  <div
    className="airline-container"
  >
    <div
      className="airline-column"
    >
      <div
        className="airline-header-container"
      >
        <AirlineHeader
          airlineAttributes={
            Object {
              "avg_score": 4.5,
              "id": "123",
              "image_url": "testAirlineImageUrl",
              "name": "testAirlineName",
              "reviews": Array [
                Object {
                  "description": "testReviewDescription",
                  "id": "456",
                  "score": 3,
                  "title": "testReviewTitle",
                },
              ],
            }
          }
        >
          <div
            className="airline-header-container"
          >
            <h1>
              <img
                alt="testAirlineName"
                className="airline-header-image"
                src="testAirlineImageUrl"
              />
              testAirlineName
            </h1>
            <div>
              <div
                className="airline-total-reviews"
              >
                4.5 out of 5
              </div>
              <div
                className="airline-rating-out-of-five"
              >
                1 user reviews
              </div>
            </div>
          </div>
        </AirlineHeader>
        <Review
          key="0"
          reviewDescription="testReviewDescription"
          reviewId="456"
          reviewScore={3}
          reviewTitle="testReviewTitle"
          setDeleteButtonClicked={[Function]}
        >
          <div
            className="review-card"
          >
            <div
              className="review-rating-container"
            >
              <img
                alt="reviewIcon"
                className="review-icon"
                src="test-file-stub"
              />
              <AirlineRating
                reviewScore={3}
              >
                <span
                  className="airline-rating-wrapper"
                >
                  <div
                    className="airline-rating-filled-in"
                  />
                  <div
                    className="airline-rating-filled-in"
                  />
                  <div
                    className="airline-rating-filled-in"
                  />
                  <div
                    className="airline-rating-not-filled-in"
                  />
                  <div
                    className="airline-rating-not-filled-in"
                  />
                </span>
              </AirlineRating>
              <DeleteReviewButton
                reviewId="456"
                setDeleteButtonClicked={[Function]}
              >
                <div
                  className="delete-review-button"
                  id="delete-review-button"
                  onClick={[Function]}
                >
                  <div
                    className="x-symbol"
                  >
                    x
                  </div>
                </div>
              </DeleteReviewButton>
            </div>
            <div
              className="review-title"
            >
              testReviewTitle
            </div>
            <div
              className="review-description"
            >
              testReviewDescription
            </div>
          </div>
        </Review>
      </div>
    </div>
    <div
      className="airline-column"
    >
      <div>
        <ReviewForm
          airlineAttributes={
            Object {
              "avg_score": 4.5,
              "id": "123",
              "image_url": "testAirlineImageUrl",
              "name": "testAirlineName",
              "reviews": Array [
                Object {
                  "description": "testReviewDescription",
                  "id": "456",
                  "score": 3,
                  "title": "testReviewTitle",
                },
              ],
            }
          }
          onReviewFormInputChange={[Function]}
          onReviewFormSubmit={[Function]}
          setReviewFormRating={[Function]}
        >
          <div
            className="review-form-container"
          >
            <form
              id="review-form"
              onSubmit={[Function]}
            >
              <div
                className="review-form-headline"
              >
                Have and experience  with testAirlineName? Share your review!
              </div>
              <div
                className="review-form-inputs-container"
              >
                <input
                  className="review-form-title-input"
                  id="review-form-title-input"
                  name="title"
                  onChange={[Function]}
                  placeholder="Review Title"
                  type="text"
                />
              </div>
              <div
                className="review-form-inputs-container"
              >
                <input
                  className="review-form-description-text-area"
                  id="review-form-description-input"
                  name="description"
                  onChange={[Function]}
                  placeholder="Review Description"
                  type="text"
                />
              </div>
              <div
                className=""
              >
                <div
                  className="review-form-rating-container"
                >
                  <div
                    className="review-form-rating-title-text"
                  >
                    Rate This Airline
                  </div>
                  <div
                    className="review-form-rating-options"
                  >
                    <input
                      checked={false}
                      className="review-form-rating-options-input"
                      id="rating-5"
                      name="rating"
                      type="radio"
                      value={5}
                    />
                    <label
                      className="review-form-rating-options-label"
                      id="review-form-rating-label"
                      onClick={[Function]}
                    />
                    <input
                      checked={false}
                      className="review-form-rating-options-input"
                      id="rating-4"
                      name="rating"
                      type="radio"
                      value={4}
                    />
                    <label
                      className="review-form-rating-options-label"
                      id="review-form-rating-label"
                      onClick={[Function]}
                    />
                    <input
                      checked={false}
                      className="review-form-rating-options-input"
                      id="rating-3"
                      name="rating"
                      type="radio"
                      value={3}
                    />
                    <label
                      className="review-form-rating-options-label"
                      id="review-form-rating-label"
                      onClick={[Function]}
                    />
                    <input
                      checked={false}
                      className="review-form-rating-options-input"
                      id="rating-2"
                      name="rating"
                      type="radio"
                      value={2}
                    />
                    <label
                      className="review-form-rating-options-label"
                      id="review-form-rating-label"
                      onClick={[Function]}
                    />
                    <input
                      checked={false}
                      className="review-form-rating-options-input"
                      id="rating-1"
                      name="rating"
                      type="radio"
                      value={1}
                    />
                    <label
                      className="review-form-rating-options-label"
                      id="review-form-rating-label"
                      onClick={[Function]}
                    />
                  </div>
                </div>
              </div>
              <button
                className="review-form-submit-button"
                type="submit"
              >
                Submit Your Review
              </button>
              <div
                className="review-form-airplane-image-container"
              >
                <img
                  alt="reviewFormAirplaneImage"
                  className="review-form-airplane-image"
                  src="test-file-stub"
                />
              </div>
            </form>
          </div>
        </ReviewForm>
      </div>
    </div>
  </div>
</Airline>
`;
